src/components/ConfirmMonthModal.js

```diff
@@ -0,0 +1,63 @@
+const ConfirmMonthModal = ({
+  setIsModalOpen,
+  setSelectedMonth,
+  monthToShow,
+}) => {
+  const handleMonthChange = () => {
+    setSelectedMonth(monthToShow);
+    setIsModalOpen(false);
+  };
+
+  return (
+    <div
+      className="modal fade show"
+      style={{
+        display: 'block',
+        backgroundColor: 'rgba(0,0,0,0.6)',
+      }}
+      id="confirmationHistoryView"
+      tabIndex="-1"
+      aria-labelledby="Select month view modal"
+      aria-hidden="true"
+    >
+      <div className="modal-dialog modal-dialog-centered">
+        <div className="modal-content">
+          <div className="modal-header">
+            <h5 className="modal-title" id="exampleModalLabel">
```

Comment:
the markup of this modal is very similar to the one above. can we somehow add an abstraction which we then reuse? e.g. via subclasses or parameterized function components.

Reply:
refactored completely!